스키마 업로드

hobulian opened the pull request:
Bridge_sns DB의 스키마 업로드

hobulian committed:
- 스키마 업로드

f-lab-TJ commented on schema/bridge_sns_member.sql:
```diff
@@ -0,0 +1,61 @@
+/*!50503 SET character_set_client = utf8mb4 */;
+CREATE TABLE `member` (
+  `id` int unsigned NOT NULL AUTO_INCREMENT,
+  `pw` varchar(20) NOT NULL,
```
패스워드를 20글자로 제한한 이유가 있을까요?
패스워드는 저장 시 어떻게 해야할까요? 관련 글을 찾아보면 도움이 될 것 같습니다.

hobulian replied:
보안상의 이유로, 패스워드는 SHA-256로 해싱해서 저장해야한다 합니다. 해시값은 64자로 반환되기 때문에, varchar(64)가 적당하다고 합니다.

f-lab-TJ commented on schema/bridge_sns_member.sql:
```diff
@@ -0,0 +1,61 @@
+/*!40101 SET @saved_cs_client     = @@character_set_client */;
+/*!50503 SET character_set_client = utf8mb4 */;
+CREATE TABLE `member` (
+  `id` int unsigned NOT NULL AUTO_INCREMENT,
```
id를 숫자 및 auto_increment로 한 이유가 있을까요?
auto_increment를 사용할때 단점은 어떤 것이 있을까요?

f-lab-TJ commented on schema/bridge_sns_post.sql:
```diff
@@ -0,0 +1,57 @@
+  `created_at` timestamp NOT NULL DEFAULT CURRENT_TIMESTAMP,
+  `modified_at` timestamp NULL DEFAULT NULL,
+  PRIMARY KEY (`id`),
+  KEY `idx_post_user_id_created_at` (`user_id`,`created_at` DESC,`id`),
```
해당 구문은 왜 필요한가요?

hobulian committed:
- member의 id를 int에서 binary(16)으로 변경
- 위의 변경사항에 맞게 member의 id를 외래키로 사용하는 다른 테이블의 구문에서도 int에서 binary(16)으로 변경
- member의 pw를 varchar(20)에서 varchar(64)로 변경

f-lab-TJ commented on schema/bridge_sns_alarm.sql:
```diff
@@ -0,0 +1,63 @@
+/*!40101 SET @saved_cs_client     = @@character_set_client */;
+/*!50503 SET character_set_client = utf8mb4 */;
+CREATE TABLE `alarm` (
+  `id` int unsigned NOT NULL AUTO_INCREMENT,
```
다른 id들도 AUTO_INCREMENT int가 적절한지 아닌지, 한번 리뷰해보도록 합시다.

f-lab-TJ commented on schema/bridge_sns_alarm.sql:
```diff
@@ -0,0 +1,63 @@
+  `user_id` binary(16) NOT NULL,
+  `type` varchar(10) NOT NULL,
+  `alarm_from` int unsigned NOT NULL,
+  `post_id` int unsigned DEFAULT NULL,
```
DEFAULT가 NULL인 이유는 뭔가요?
